### PR TITLE
Make mantine colors compatible with v7

### PR DESCRIPTION
**Pull request by ranquild**
Context: https://metaboat.slack.com/archives/C057WD5L0JG/p1692612437317219
[Mantine 7.0 changelog](https://v7.mantine.dev/changelog/7-0-0)

Changes:
- Always 10 theme colors per color name
- Move focus to its own name to avoid confusion with brand colors
- Remove `theme.fn.primaryColor()` since it was removed in v7

How to verify:
- `yarn storybook`
- Mantine components should look normal

### CODE REVIEW

File: frontend/src/metabase/ui/theme.ts [hunk 1]
```diff
@@ -12,12 +12,30 @@ import {
   getTitleOverrides,
 } from "./components";
 
+type ThemeColors = MantineTheme["colors"]["brand"];
+
+const getThemeColors = (colors: string[]): ThemeColors => {
```
Comment (WiNloSt) [inline frontend/src/metabase/ui/theme.ts:17]: I'm not sure how the new 10 color shades would work in Mantine. Seems like we're filling undefined color with `transparent` instead as if we don't care about them.

Do you think we'll eventually generate 10 shades per color as Mantine 7 does? Or are we going to keep using predefined shades and ignore the 10 shades from Mantine. 

Also, it has a [color generator](https://v7.mantine.dev/changelog/7-0-0#colors-generator) which generates 10 shades of color based on a single input color that could be helpful.

**Review by WiNloSt — COMMENTED** [review]
Left a comment about Mantine colors, otherwise, looks good to me.

File: frontend/src/metabase/ui/theme.ts (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/ui/theme.ts:17]: I think we explicitly don't want to have 10 shades as in Figma there are only few; other shades shouldn't be available at all

**Review by WiNloSt — APPROVED** [review]
Looks good to me.

**Pull request merged**